### ISSUE TITLE
Should be able to store a Value in a struct

### ISSUE THREAD

**Issue by andrewrynhard** [dtolnay/serde-yaml#23]
In the process of moving from `serde_json` to `serde_yaml` and I am trying to store a `Value` in a struct
```rust
use serde_yaml::Value;

#[derive(Debug, Serialize, Deserialize)]
pub struct GenericInstructions {
    #[serde(rename="type")]
    pub _type: String,
    pub config: Value,
}
```
but I get
```
the trait bound `serde_yaml::Yaml: serde::Serialize` is not satisfied
```

This works fine using `serde_json`.

**Comment by Amndeep7** [comment]
I think this can be generalized to `Value` having to implement the `Serialize` and `Deserialize` traits.  Moreover, this would create parity with `serde_json`'s behavior in regards to the "serialize-ability" of `Value`: https://docs.serde.rs/serde_json/value/enum.Value.html.

**Comment by dtolnay** [comment]
I released [0.5.0](https://github.com/dtolnay/serde-yaml/releases/tag/0.5.0) containing `serde_yaml::Value` which implements Serialize and Deserialize.
